backend/pkg/api/data_access/vdb_summary.go

```diff
@@ -733,62 +739,138 @@ func (d *DataAccessService) GetValidatorDashboardGroupSummary(ctx context.Contex
 	return ret, nil
 }
 
-func (d *DataAccessService) internal_getElClAPR(ctx context.Context, validators []t.VDBValidator, days int) (elIncome decimal.Decimal, elAPR float64, clIncome decimal.Decimal, clAPR float64, err error) {
-	var reward sql.NullInt64
+func (d *DataAccessService) internal_getElClAPR(ctx context.Context, dashboardId t.VDBId, groupId int64, days int) (elIncome decimal.Decimal, elAPR float64, clIncome decimal.Decimal, clAPR float64, err error) {
 	table := ""
 
 	switch days {
 	case 1:
-		table = "validator_dashboard_data_rolling_daily"
+		table = "validator_dashboard_data_rolling_24h"
 	case 7:
-		table = "validator_dashboard_data_rolling_weekly"
+		table = "validator_dashboard_data_rolling_7d"
 	case 30:
-		table = "validator_dashboard_data_rolling_monthly"
+		table = "validator_dashboard_data_rolling_30d"
 	case -1:
 		table = "validator_dashboard_data_rolling_90d"
 	default:
 		return decimal.Zero, 0, decimal.Zero, 0, fmt.Errorf("invalid days value: %v", days)
 	}
 
-	query := `select (SUM(COALESCE(balance_end,0)) + SUM(COALESCE(withdrawals_amount,0)) - SUM(COALESCE(deposits_amount,0)) - SUM(COALESCE(balance_start,0))) reward FROM %s WHERE validator_index = ANY($1)`
+	type RewardsResult struct {
+		EpochStart     uint64        `db:"epoch_start"`
+		EpochEnd       uint64        `db:"epoch_end"`
+		ValidatorCount uint64        `db:"validator_count"`
+		Reward         sql.NullInt64 `db:"reward"`
+	}
 
-	err = db.AlloyReader.GetContext(ctx, &reward, fmt.Sprintf(query, table), validators)
-	if err != nil || !reward.Valid {
+	var rewardsResultTable RewardsResult
+	var rewardsResultTotal RewardsResult
+
+	rewardsDs := goqu.Dialect("postgres").
+		Select(
+			goqu.L("MIN(epoch_start) AS epoch_start"),
+			goqu.L("MAX(epoch_end) AS epoch_end"),
+			goqu.L("COUNT(*) AS validator_count"),
+			goqu.L("(SUM(COALESCE(r.balance_end,0)) + SUM(COALESCE(r.withdrawals_amount,0)) - SUM(COALESCE(r.deposits_amount,0)) - SUM(COALESCE(r.balance_start,0))) AS reward")).
+		From(goqu.L(fmt.Sprintf("%s AS r FINAL", table)))
+
+	if len(dashboardId.Validators) > 0 {
+		rewardsDs = rewardsDs.
+			Where(goqu.L("validator_index IN ?", dashboardId.Validators))
+	} else {
+		rewardsDs = rewardsDs.
+			InnerJoin(goqu.L("users_val_dashboards_validators v"), goqu.On(goqu.L("r.validator_index = v.validator_index"))).
+			Where(goqu.L("v.dashboard_id = ?", dashboardId.Id))
```

Comment:
this is prob slow as it isn't using the `with` subquery optimization that gets used above

Reply:
I changed it to the subquery for here and also the validator slashing list which should now be all clickhouse queries.
However I also removed the WHERE condition after the INNER JOIN since it was redudant, or was there an unknown performance trick that needed it?

Reply:
> However I also removed the WHERE condition after the INNER JOIN since it was redudant, or was there an unknown performance trick that needed it?

doesnt seem like the clickhouse query planner is smart enough to optimize the join that way. without the where filter the query always scans the complete `from` table

Reply:
Ah, I see. I reverted the removal of the WHERE.